pkg/kubelet/status_manager.go

```diff
@@ -135,13 +136,29 @@ func (s *statusManager) syncBatch() error {
 	}
 	// TODO: make me easier to express from client code
 	statusPod, err = s.kubeClient.Pods(statusPod.Namespace).Get(statusPod.Name)
+	if errors.IsNotFound(err) {
+		glog.V(3).Infof("Pod %q was deleted on the server", pod.Name)
+		return nil
+	}
 	if err == nil {
 		statusPod.Status = status
-		_, err = s.kubeClient.Pods(pod.Namespace).UpdateStatus(statusPod)
 		// TODO: handle conflict as a retry, make that easier too.
+		statusPod, err = s.kubeClient.Pods(pod.Namespace).UpdateStatus(statusPod)
 		if err == nil {
 			glog.V(3).Infof("Status for pod %q updated successfully", pod.Name)
-			return nil
+
+			if statusPod.DeletionTimestamp == nil {
```

Comment:
Ok, will test with that.

----- Original Message -----
> >  		if err == nil {
> >  			glog.V(3).Infof("Status for pod %q updated successfully", pod.Name)
> > -			return nil
> > +
> > +			if statusPod.DeletionTimestamp == nil {
> 
> Would it be better to check `pod.DeletionTimestamp` instead here?
> `pod` was what the worker saw when syncing. If the `DeletionTimestamp` was
> set at the time of syncing, we can be certain that the pod would never run
> again. On the other hand, `statusPod` is the latest view of the pod from the
> apiserver. If the pod worker lags behind and still doesn't see the non-nil
> timestamp, it might try to start the pod in the next run.  It's rare but I
> think it could happen.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9165/files#r31831882